### DISCUSSION TITLE
RegEx is bad for parsing routes

### DISCUSSION THREAD

**Discussion by levovix0** [HapticX/happyx#243] [Ideas]
I was developing a relatively large website using happyx. At some point, compiling the server code began to take several minutes, and the binary took up a lot of space (around 16 megabytes). I looked into the code generated by the Nim compiler. Most of the server's code was taken up by regular expressions. Then I tried to implement the same functionality (routing to about 40 different urls) using `url.split('/')` and a small macro. The compilation speed increased 15 times, and the binary began to weigh 3 times less. I haven't measured the parsing speed, but I don't think it's any more significant than using regular expressions. Is it possible to somehow implement routing in happyx using a simpler method than RegEx?

**Comment by Ethosa** [comment]
@levovix0 can you give me some examples?

**Reply by levovix0** [reply]
```nim
import happyx

serve "localhost", 6015:
  get "/api/account/{userid:int}":
    ## ...
```
generates
```nim
# ...
case req.reqMethod
of HttpGet:
  if contains(urlPath, re2("^/api/account/(\\d+)$")):
    let userid = parseInt(decodeUrl(urlPath[
        group(findAll(urlPath, re2("^/api/account/(\\d+)$"), 0)[0], 0)],
        true))
    ## ...
# ...
```
result in 25000 lines of c code and a 2.1 megabytes binary (1.3 MB with -d:danger)

but [this](https://gist.github.com/levovix0/539bebcf6f3e7dd67ef154cb1ad97ab5), for example
generates
```nim
# ...
var userid: int
if httpMethod == HttpMethod(1) and len(url) == 3 and url[0] == "api" and url[1] == "account" and
    (try: (userid = parseToType(url[2], int); true) except: return):
  ## ...
# ...
```
result in 2700 lines of c code and a 700 kilobytes binary (300 KB with -d:danger)
